galaxy_ng/app/tasks/deletion.py

```diff
@@ -34,8 +35,13 @@ def delete_container_distribution(instance_ids):
     log.info("Running core.general_multi_delete to delete distro and repo")
     general_multi_delete(instance_ids=instance_ids)
 
+    repository_ids = [item[0] for item in instance_ids]
+    repository_content_pks = \
+        RepositoryContent.objects.filter(repository__pk__in=repository_ids)\
```

Comment:
This also needs to exclude content that is in other repositories. I _think_ this should work, but you'll need to test it.

```
RepositoryContent.objects.annotate(num_repos=Count("repository")).filter(repository__pk__in=repository_ids, num_repos=1)
```